src/platform/Makefile.am

```diff
@@ -107,6 +107,7 @@ noinst_HEADERS                          = \
     @top_srcdir@/src/platform/ESP32/CHIPDevicePlatformConfig.h \
     @top_srcdir@/src/platform/ESP32/ConnectivityManagerImpl.h \
     @top_srcdir@/src/platform/ESP32/BLEManagerImpl.h \
+    @top_srcdir@/src/platform/ESP32/nimble/BLEManagerImpl.h \
```

Comment:
Can we add src/platform/bluedroid directory too and move the current BLEManagerImpl.h and BLEManagerImpl.cpp there ?

Also can we use CONFIG_BT_BLUEDROID_ENABLED / CONFIG_BT_NIMBLE_ENABLED to conditionally add those files to the build 
There reason I'm asking is because in the current shape even if I enabled NimBLE in menuconfig it stills try to build the current BLEManagerImpl.h/cpp files related to Bluedroid.